Core/Src/bms.c

```diff
@@ -40,25 +42,43 @@ void initBmsGopherCan(CAN_HandleTypeDef* hcan)
 	// initialize CAN
 	if (init_can(GCAN0, hcan, BMS_ID, BXTYPE_MASTER))
 	{
-		gBms.bmsHwState = BMS_GSNS_INIT_FAILURE;
+		gBms.bmsHwState = BMS_GSNS_FAILURE;
 	}
 }
 
 /*!
   @brief   Initialization function for the battery pack
   @param   numBmbs - The expected number of BMBs in the daisy chain
+  @returns bool True if initialization successful, false otherwise
 */
-void initBatteryPack(uint32_t numBmbs)
+bool initBatteryPack(uint32_t* numBmbs)
 {
 	Bms_S* pBms = &gBms;
 
-	pBms->numBmbs = NUM_BMBS_PER_PACK;
+	if (!initASCI())
+	{
+		return false;
+	}
 
-	memset (pBms, 0, sizeof(Bms_S));
+	if (!helloAll(numBmbs))
+	{
+		return false;
+	}
 
-	initBmbs(numBmbs);
+	if (*numBmbs != NUM_BMBS_PER_PACK)
+	{
+		Debug("Number of BMBs detected (%lu) doesn't match expectation (%d)\n", *numBmbs, NUM_BMBS_PER_PACK);
+		return false;
```

Comment:
Do we need to keep this check? Can we just use helloall to find numbmbs and use whatever it returns? That way it automatically scales to however many bmbs we have in the chain.

Reply:
What if one BMB becomes disconnected? How do we handle that? Or if for whatever reason we have one extra BMB. Would we try to write into memory that hasn't been initialized since the BMS data struct only initializes enough bmb structs for NUM_BMBS_PER_PACK?